src/vs/workbench/browser/parts/editor/editorPart.ts

```diff
@@ -240,7 +241,18 @@ export class EditorPart extends Part implements IEditorPart, IEditorGroupService
 		// while the UI is not yet ready. Clients have to deal with this fact and we have to make sure that the
 		// stacks model gets updated if any of the UI updating fails with an error.
 		const group = this.ensureGroup(position, !options || !options.preserveFocus);
-		const pinned = !this.previewEditors || (options && (options.pinned || typeof options.index === 'number')) || input.isDirty();
+		let pinned = !this.previewEditors || (options && (options.pinned || typeof options.index === 'number')) || input.isDirty();
+		if (this.previewEditors) {
+			pinned = !this.configurationService.getConfiguration<IWorkbenchEditorConfiguration>().workbench.editor.enablePreviewFromQuickOpen;
```

Comment:
@wprater you can use most service from most classes by just adding it to the constructor as I was showing it for `quickOpenController`. We use constructor service injection and you typically do not have to deal with it.

I am not yet sure how we can get this working without introducing a new editor option `forcePreview` (or something like that) to open editors non-pinned when navigating in quick open.

I see your idea with using `enablePreviewFromQuickOpen` setting but I think this is different: it controls if a file opened from quick open shows up pinned or as preview. In our case we always want preview no matter what this setting does.